[Runtime] Add an option to produce non-symbolicated backtraces.

al45tair opened the pull request:
Symbolication can take some time, depending on the binaries involved. In certain contexts it's better for the backtrace to finish quickly, and then symbolication could be done offline.

rdar://122302117

al45tair committed:
[Runtime] Add an option to produce non-symbolicated backtraces.

Symbolication can take some time, depending on the binaries involved.
In certain contexts it's better for the backtrace to finish quickly,
and then symbolication could be done offline.

rdar://122302117

al45tair commented:
@swift-ci Please smoke test

al45tair committed:
[Docs] Updated Backtracing docs with new symbolicate setting.

Document the new `symbolicate` setting.

rdar://122302117

al45tair commented:
@swift-ci Please smoke test

mikeash commented on stdlib/public/libexec/swift-backtrace/TargetMacOS.swift:
```diff
@@ -260,24 +266,32 @@ class Target {
   }
 
-  public func redoBacktraces(limit: Int?, top: Int, cache: Bool) {
+  public func redoBacktraces(limit: Int?, top: Int,
```
No need to do it now, but it might be nice to pull out these options into some sort of options struct so future changes don't have to be repeated all over the place.

al45tair replied:
Yeah, I think the entire thing could do with a tidy up, honestly.

al45tair replied:
I think this is probably a job for a separate PR though.

mikeash commented on stdlib/public/libexec/swift-backtrace/main.swift:
```diff
@@ -202,6 +203,10 @@ Generate a backtrace for the parent process.
                         structure.  <addr> should be in hexadecimal.
+
+--symbolicate [<bool>]  Set whether or not to resolve addresses into symbols.
+                        Also controls whether or not we look for inline frame
```
This is a bit vague about what that control looks like, although it's fairly obvious. Maybe say something like, "When symbolication is disabled, inline frame data is also not used and the backtrace only shows the actual stack frame return addresses." Except better, because I don't know how to word it well.

al45tair replied:
I've updated the text, but that's partly because I added another option :-)

mikeash replied:
Looks good!

al45tair committed:
[Runtime] Improved symbolication option for backtracing.

Rather than just on or off, I've changed it to allow "off", "fast",
or "full".  "fast" means that we'll do symbol lookup, but we won't
try to find inline frames and we won't run line number programs
(those are the things that are taking considerable time in some
cases).

rdar://122302117

al45tair commented:
@swift-ci Please smoke test
